lib/recipient.rb

```diff
@@ -0,0 +1,58 @@
+require 'dotenv'
+require 'json'
+
+# require_relative '../lib/bot_settings.json'
+Dotenv.load
+
+class SlackApiError < StandardError; end
+class NoMessageError < StandardError; end
+
+class Recipient
+  attr_reader :id, :name
+
+  def initialize(id, name)
+    @id = id
+    @name = name
+  end
+
+  API_KEY = ENV["SLACK_API_TOKEN"]
+  def self.get(url)
+    response = HTTParty.get(url, query: { token: API_KEY })
+
+    raise SlackApiError.new(response["error"]) unless response.parsed_response["ok"]
+
+    return response
+  end
+
+  CHAT_URL = "https://slack.com/api/chat.postMessage"
+  BOT_API_KEY = ENV["SLACK_BOT_API_TOKEN"]
```

Comment:
You should use a single token for getting and sending.

```suggestion
  BOT_API_KEY = ENV["SLACK_TOKEN"]
```